Sources/Downloader/DefaultDownloader.swift

```diff
@@ -327,7 +342,14 @@ extension DefaultDownloader: URLSessionDownloadDelegate {
     }
     
     func urlSession(_ session: URLSession, downloadTask: URLSessionDownloadTask, didWriteData bytesWritten: Int64, totalBytesWritten: Int64, totalBytesExpectedToWrite: Int64) {
-        self.delegate?.downloader(self, didProgress: bytesWritten)
+        // Only report progress updates if the throttle permits
+        guard isProgressUpdatePermitted() else {
+            throttledBytesWritten += bytesWritten
+            return
+        }
+        self.delegate?.downloader(self, didProgress: bytesWritten + throttledBytesWritten)
+        lastProgressRefreshTime = Date()
+        throttledBytesWritten = 0
```

Comment:
In the `urlSession(_ session: URLSession, downloadTask: URLSessionDownloadTask, didFinishDownloadingTo location: URL)` delegate, add a call to `self.delegate?.downloader(self, didProgress: throttledBytesWritten)` before doing anything else (before the do..catch block).

Reply:
I suggest on adding a check if there is 'throttledBytesWritten' to report before calling to didProgress.

Reply:
@x-NR-x Agreed. Added an additional check in this case that ensures that 'throttledBytesWritten' is non-zero.